core/middleware.ts

```diff
@@ -1,10 +1,17 @@
 import { composeMiddlewares } from './middlewares/compose-middlewares';
+import { withAnalyticsCookies } from './middlewares/with-analytics-cookies';
 import { withAuth } from './middlewares/with-auth';
 import { withChannelId } from './middlewares/with-channel-id';
 import { withIntl } from './middlewares/with-intl';
 import { withRoutes } from './middlewares/with-routes';
 
-export const middleware = composeMiddlewares(withAuth, withIntl, withChannelId, withRoutes);
+export const middleware = composeMiddlewares(
+  withAnalyticsCookies,
+  withAuth,
+  withIntl,
+  withChannelId,
+  withRoutes,
+);
```

Comment:
Since middleware runs in order and the `withIntl` stuff does some interesting rewrites/redirects depending on the config, can we add this middleware either directly before or after `withChannelId`? It shouldn't matter before or after, just that it's after `withIntl`.